### PR TITLE
Implement Optional Pagination for get_guilds

### PR DESCRIPTION
**Pull request by tylerd008**
Fix for issue #1375.

### CODE REVIEW

**Comment by Zaitara** [comment]
You need to target the `next` branch as it is a breaking change

File: src/model/user.rs [hunk 1]
```diff
@@ -192,7 +192,9 @@ impl CurrentUser {
             let mut pagination = http
                 .as_ref()
                 .get_guilds(
-                    &GuildPagination::After(guilds.last().map_or(GuildId(1), |g: &GuildInfo| g.id)),
+                    Some(&GuildPagination::After(
```
Comment (Dinnerbone) [inline src/model/user.rs]: I think this should be changed to `None`, as it's effectively trying to workaround this very issue.

**Comment by tylerd008** [comment]
Oh whoops sorry, changed.

**Comment by drklee3** [comment]
Maybe you should `cherry-pick` your commits onto a new branch based on `next` so you don't get all the other commits from `current` added in.

File: src/model/user.rs (hunk 1 above)
Comment (tylerd008) [reply src/model/user.rs]: Updated

**Comment by tylerd008** [comment]
> Maybe you should `cherry-pick` your commits onto a new branch based on `next` so you don't get all the other commits from `current` added in.

Oh yeah sorry let me do that real quick

**Pull request closed without merging**